docs/user-guide/102-policy-syntax.md

```diff
@@ -0,0 +1,130 @@
+be referenced with ```capture.[id]```
+
+The value from the map is a capture entry expression that evaluate the current 
+network state and store the result.
+
+The following is a semi-formal definition of the capture entry expression:
+
+```
+<letter> ::= "A" | "B" | "C" | "D" | "E" | "F" | "G"
+       | "H" | "I" | "J" | "K" | "L" | "M" | "N"
+       | "O" | "P" | "Q" | "R" | "S" | "T" | "U"
+       | "V" | "W" | "X" | "Y" | "Z" | "a" | "b"
+       | "c" | "d" | "e" | "f" | "g" | "h" | "i"
+       | "j" | "k" | "l" | "m" | "n" | "o" | "p"
+       | "q" | "r" | "s" | "t" | "u" | "v" | "w"
+       | "x" | "y" | "z"
+<digit> ::= [0-9]
+<number> ::= <digit>+
+<identifier> ::= <letter> ( <digit> | "-" | <letter> )*
+<dot> ::= "."
+<path> ::= <identifier> ( <dot> <identity> )*
+<string> ::= "\"" (<all characters>)* "\""
+<bool> ::= "true" | "false"
+
+<captureid> ::= <identity>
+<capturepath> ::= "capture" <dot> <captureid> <path>
+<eqoperator> ::= "=="
+<eqexpression> ::= <path> <eqoperator> (<string> | <number> | <bool> | <capturepath>)
+<eqoperator> ::= ":="
+<replaceexpression> ::= <path> <replaceoperator> (<string> | <number> | <bool> | <capturepath>)
+<expression> ::= <eqfilterexpression> | <replaceexpression>
+<pipe> ::= "|"
+<pipedexpression> ::= ( <capturepath> | <expression> ) <pipe> <expression>
+```
+
+### Path ```<path>```
+The path expression contains different "steps" separated by dots, each "step" 
+can be a key from a pam or a idx from a list.
+```
+interfaces.name
+routes.running.0.next-hop-interface
+```
+
+A special path is the one referencing a capture entry for that the "capture"
+reserved word has to be used:
+```
+capture.default-gw.routes.running.0
+capture.primary-nic.interfaces.0.name
+```
+
+### Equality filter ```<eqexpression>```
+Filter the current state based on specific state values. 
+The filter follows a simple syntax, similar to jsonpath. 
+Values may be explicit or appear as references to other expressions. 
+The resulting output is a full NMState state containing only the filtered values.
+```
+interfaces.name == "eth1"
+routes.destination == "0.0.0.0/0"
+dns.server == "192.168.1.1"
+interfaces.ipv4.enabled == true
+interfaces.name == capture.default-gw.interfaces.0.name
+```
+
+### Replace ```<replaceexpression>```
+These commands can replace values from the specified fields 
+at the input NMState and they can reference other capture entries.
+```
+routes.running.next-hop-interface:="br1"
+routes.running.absent:=true
+```
+
+### Pipe ```<pipexpression>```
+When expressions are piped the output from the left expression is passed 
+to the input of the right command.
+```
+capture.base-iface-routes | routes.running.next-hop-interface:="br1"
+capture.base-iface-routes | routes.running.absent:=true
+```
+
+## Desired state syntax
+
+This will follow the [NMState](https://nmstate.io/examples.html) syntax and will include optionally some reference 
```

Comment:
The state follows ...

Also, please drop "some" and use plural for `references`.